test/utils/testconfig/testconfig.go

```diff
@@ -0,0 +1,274 @@
+		GlobalConfig.Status = InitStatus
+		return GlobalConfig
+	}
+}
+func GetFullDiscoveredConfig(namespace string, aClient *testclient.ClientSet, forceUpdate bool) TestConfig {
+	defer logrus.Infof("Current PTP test config=%s", &GlobalConfig)
+	if GlobalConfig.Status == DiscoveryFailureStatus ||
+		GlobalConfig.Status == DiscoverySuccessStatus && !forceUpdate {
+		return GlobalConfig
+	}
+
+	discoverPTPConfiguration(namespace, aClient)
+	return GlobalConfig
+}
+
+// Returns the slave node label to be used in the test
+func discoverPTPConfiguration(namespace string, aClient *testclient.ClientSet) {
+	var slaves []ptpv1.PtpConfig
+	var masters []ptpv1.PtpConfig
+
+	configList, err := aClient.PtpConfigs(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting ptpconfig list, err=%s", err)
+	}
+	for _, config := range configList.Items {
+		for _, profile := range config.Spec.Profile {
+			if IsPtpMaster(*profile.Ptp4lOpts, *profile.Phc2sysOpts) {
+				masters = append(masters, config)
+			}
+			if IsPtpSlave(*profile.Ptp4lOpts, *profile.Phc2sysOpts) {
+				slaves = append(slaves, config)
+			}
+		}
+	}
+
+	//GlobalConfig.DiscoveredMasterPtpConfig=checkPtpProfileLabels(masters)
+	discoverMode(slaves, masters, aClient)
+}
+
+func discoverMode(slaves, masters []ptpv1.PtpConfig, aClient *testclient.ClientSet) {
+
+	if len(slaves) == 0 {
+		logrus.Warnf("No Configs present, cannot discover")
+		return
+	}
+	numBc := 0
+	pickedMasterConfig := checkPtpProfileLabels(masters, aClient)
+	GlobalConfig.DiscoveredMasterPtpConfig = pickedMasterConfig
+	for _, slave := range slaves {
+		pickedSlaveConfig := checkSinglePtpProfileLabels(slave, aClient)
+		if pickedSlaveConfig.Config.Name != "" {
+
+			masterIf := len(ptpv1.GetInterfaces(pickedSlaveConfig.Config, ptpv1.Master))
+			slaveIf := len(ptpv1.GetInterfaces(pickedSlaveConfig.Config, ptpv1.Slave))
+			GlobalConfig.Status = DiscoveryFailureStatus
+			// OC
+			if masterIf == 0 && slaveIf == 1 {
+				GlobalConfig.PtpModeDiscovered = OrdinaryClock
+				GlobalConfig.Status = DiscoverySuccessStatus
+				GlobalConfig.DiscoveredSlavePtpConfig = pickedSlaveConfig
+			}
+			// BC and Dual NIC BC
+			if masterIf > 1 && slaveIf == 1 {
+				if numBc == 0 {
+					GlobalConfig.DiscoveredSlavePtpConfig = pickedSlaveConfig
+				}
+				if numBc == 1 {
+					GlobalConfig.DiscoveredSlavePtpConfigSecondary = pickedSlaveConfig
+				}
+				numBc++
+			}
+		} else {
+			GlobalConfig.Status = DiscoveryFailureStatus
+		}
+	}
+	if numBc == 1 {
+		GlobalConfig.PtpModeDiscovered = BoundaryClock
+		GlobalConfig.Status = DiscoverySuccessStatus
+	}
+	if numBc == 2 {
+		GlobalConfig.PtpModeDiscovered = DualNICBoundaryClock
+		GlobalConfig.Status = DiscoverySuccessStatus
+	}
+
+}
+
+func IsPtpSlave(ptp4lOpts string, phc2sysOpts string) bool {
+	return strings.Contains(ptp4lOpts, "-s") && strings.Count(phc2sysOpts, "-a") == 1 && strings.Count(phc2sysOpts, "-r") == 1
```

Comment:
for dual ptp , in one of the slave ptpconfig (nic 2). There won't be a  phc2sysOpts metadata.